### ISSUE TITLE
CamelCaseIsOutdated

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/contributing#31]
It's a hard truth to face, but the modern C++ community is moving away from camelCase since several years. And let's be honest, abbreviations were never great in it: `doVSWorkaround`, `LWFAExample`, ...

If you want to maintain our street credibility, we will have to switch to `more_readable_names` as well.

Btw, the `STL` always did `_`.

Hopefully there is a Clang-tool assisting us in the transition ;-)

**Comment by j-stephan** [comment]
Ping! Let us get rid of CamelCase!

**Comment by bernhardmgruber** [comment]
I do not think that there is a trend in the modern C++ community to move away from camelCase/PascalCase to snake_case. I like to believe that most communities just stick to what they are used to or what their companies empose on them. So @ax3l I would be interested in the source of your claim ;)

Nevertheless, I am fine with most styles, so feel free to pick one.

For the transition, clang-tidy seems to be your friend, including automatic fixits:
https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html

**Comment by psychocoderHPC** [comment]
> Ping! Let us get rid of CamelCase!

I am old, I love camelCase names^^

**Comment by j-stephan** [comment]
How old are you? snake_case was already used by Kernighan & Ritchie in "The C Programming Language".

**Comment by j-stephan** [comment]
If we are agreeing on using clang-format: Is there anything that stops us from reformatting to the C++ Core Guidelines? http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

**Comment by ax3l** [comment]
@bernhardmgruber My sources are primarily folks tweeting from the ISO C++ committee and our friends from QuantStack ([xtensor](https://github.com/xtensor-stack/xtensor), et al.). I mean the std libs are snake case (already good enough for me), modern frameworks such as SYCL are snake case, CUDA is old and clutchAllTogetherCase - new CUDA goes straight ISO C++ and consequently snake case, too. We were young and tried to save_all_this_space, happened to Fortran programmers as well back in the day.

I personally and totally subjectively think there is generally a trend to snake_case in new C++ libraries without having seen a meta-analysis on it yet. Now that I am getting older, I feel it is more readable and goes coincidentally great with side-by-side comparisons to Python when doing pybind11 bindings for C++ libs.

> If we are agreeing on using clang-format: Is there anything that stops us from reformatting to the C++ Core Guidelines? http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

Sounds great :)

**Comment by j-stephan** [comment]
Since we now have a clang-tidy file which manages the naming style this issue can be closed I think. (Although we still have CamelCase, ugh...)
